### PR TITLE
#123: 🔥  use either syntax compatible with scala 2.11 (closes #123)

### LINKED ISSUE #123: 🔥  use either syntax compatible with scala 2.11

### PR DESCRIPTION
**Pull request by calippo**
Closes #123

### tests performed
- unit tests should not break up

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -32,14 +33,15 @@ val commonDependencies = Seq(
...
+  pureConfig,
+  cats
 ) ++ circeDependencies ++ loggingBackendDependencies
 
 lazy val commonSettings = Seq(
   bintrayOrganization := Some("buildo"),
   organization := "io.buildo",
   licenses += ("MIT", url("https://github.com/buildo/wiro/blob/master/LICENSE")),
-  scalaVersion := "2.12.4",
+  scalaVersion := "2.11.8",
```
Comment (gabro) [inline build.sbt]: since you're at it, you can use `2.11.12`

File: serverAkkaHttp/src/main/scala/RPCRouter.scala [hunk 2]
```diff
@@ -79,11 +79,10 @@ trait Router extends RPCServer with PathMacro with MetaDataMacro with LazyLoggin
...
     }
 
-  private[this] def parseJsonObject(s: String): Either[ParsingFailure, Json] =
-    parse(s).filterOrElse(
-      p = (json: Json) => json.isObject,
-      zero = ParsingFailure("The parsed Json is not an object", new Exception())
-    )
+  private[this] def parseJsonObject(s: String): Either[ParsingFailure, Json] = {
+    val failure = ParsingFailure("The parsed Json is not an object", new Exception())
+    parse(s).ensureOr(_ => failure)((json: Json) => json.isObject)
```
Comment (gabro) [inline serverAkkaHttp/src/main/scala/RPCRouter.scala]:
```scala
parse(s).ensure(failure)(_.isObject)
```

**Review by gabro — CHANGES_REQUESTED** [review]

**Comment by gabro** [comment]
@calippo I don't know what's wrong with the CI. I've restarted it 3 times and it keeps failing with a timeout. Can you check?

**Comment by calippo** [comment]
@gabro fixed changing default timeout

**Review by gabro — APPROVED** [review]

**Pull request merged**